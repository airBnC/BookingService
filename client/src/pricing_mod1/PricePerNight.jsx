import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Holder = styled.div`
  width: inherit;
  justify-content: flex-start;
`;
export const Price = styled.span`
  font-size: 30px;
`;
export const Msg = styled.span`
  margin-left: 5px;
`;

export default function PricePerNight( props ) {
  PricePerNight.propTypes = {
    price: PropTypes.number,
  };
  PricePerNight.defaultProps = {
    price: 0,
  };
  if ( props.price ) {
    return (
      <Holder>
        <Price>
          ${props.price}
        </Price>
        <Msg>
          per night
        </Msg>
      </Holder>
    );
  }
  return null;
}
