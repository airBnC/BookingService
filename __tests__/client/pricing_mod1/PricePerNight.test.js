import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import 'jest-styled-components';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import PricePerNight, { Price } from '../../../client/src/pricing_mod1/PricePerNight';

configure( { adapter: new Adapter() } );

const props = {
  price: Math.floor( Math.random() * 1000 ),
  priceNull: undefined,
};

describe( 'Price Per Nights Component', () => {
  it( 'renders price correctly', () => {
    const tree = renderer
      .create( <Price> {props.price}</Price> )
      .toJSON();
    const expected = [ ' ', `${ props.price }` ];
    expect( tree.children )
      .toEqual( expect.arrayContaining( expected ) );
  } );
  it( 'does not render if props.price is undefined', () => {
    const tree = renderer
      .create( <PricePerNight price={props.priceNull} /> )
      .toJSON();
    expect( tree ).toMatchSnapshot();
  } );
} );
