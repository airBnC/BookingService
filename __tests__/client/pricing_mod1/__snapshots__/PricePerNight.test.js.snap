// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Per Nights Component does not render if props.price is undefined 1`] = `null`;
